File: TsinghuaNet/ConnectHelper.cs
using System.Net.Http;

namespace TsinghuaNet
{
    public enum NetState
    {
        Unknown,
        Net,
        Auth4,
        Auth6
    }

    public static class ConnectHelper
    {
        public static IConnect GetHelper(NetState state) => GetHelper(state, null, null, null);
        public static IConnect GetHelper(NetState state, HttpClient client) => GetHelper(state, null, null, client);
        public static IConnect GetHelper(NetState state, string username, string password) => GetHelper(state, username, password, null);
        public static IConnect GetHelper(NetState state, string username, string password, HttpClient client)
        {
            switch (state)
            {
                case NetState.Net:
                    return new NetHelper(username, password, client);
                case NetState.Auth4:
                    return new Auth4Helper(username, password, client);
                case NetState.Auth6:
                    return new Auth6Helper(username, password, client);
                default:
                    return null;
            }
        }
    }
}

File: TsinghuaNet.Helper/Helpers/ConnectHelper.vb
Imports System.Net.Http

Public Enum NetState
    Unknown
    Net
    Auth4
    Auth6
End Enum

Public Module ConnectHelper
    Public Function GetHelper(state As NetState) As IConnect
        Return GetHelper(state, Nothing, Nothing, Nothing)
    End Function

    Public Function GetHelper(state As NetState, client As HttpClient) As IConnect
        Return GetHelper(state, Nothing, Nothing, client)
    End Function

    Public Function GetHelper(state As NetState, username As String, password As String) As IConnect
        Return GetHelper(state, username, password, Nothing)
    End Function

    Public Function GetHelper(state As NetState, username As String, password As String, client As HttpClient) As IConnect
        Select Case state
            Case NetState.Net
                Return New NetHelper(username, password, client)
            Case NetState.Auth4
                Return New Auth4Helper(username, password, client)
            Case NetState.Auth6
                Return New Auth6Helper(username, password, client)
            Case Else
                Return Nothing
        End Select
    End Function
End Module
